pkg/build/kube/files.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+const kubeletService = `[Unit]
+Description=kubelet: The Kubernetes Node Agent
+Documentation=http://kubernetes.io/docs/
+
+[Service]
+ExecStart=/usr/bin/kubelet
+Restart=always
+StartLimitInterval=0
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+`
+
+const tenKubeadmConf = `# Note: This dropin only works with kubeadm and kubelet v1.11+
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/default/kubelet
+ExecStart=
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_CONFIG_ARGS $KUBELET_KUBEADM_ARGS $KUBELET_EXTRA_ARGS
```

Comment:
baking in the unit/drop-in is not ideal. because kind now has to maintain and sync with the upstream systemd files...
we have deb/rpm specs scattered in k/k and k/release. kind *probably* shouldn't manage it's own copies.

i would HTTP download them from github:
https://github.com/kubernetes/release/blob/master/debian/xenial/kubelet/lib/systemd/system/kubelet.service
https://github.com/kubernetes/release/blob/master/debian/xenial/kubeadm/channel/stable/etc/systemd/system/kubelet.service.d/post-1.10/10-kubeadm.conf

(1.15 is hopefully going to nuke the above, though).

Reply:
Hi @neolit123,

I was just following @BenTheElder's `// TODO` from the BazelBits code. Still, I've no issue downloading them either. Ben?

Reply:
Hi @neolit123,

I've addressed your feedback here. I first try to get the files at the remote URIs, and if that fails I default back to the constant values.

Reply:
waiting on Ben's comments on this one. pulling from the release repo is not ideal too.

Reply:
Hi @neolit123,

Sounds good. I'm happy to take this in any direction. I figured the constants were what @BenTheElder was thinking since Kind already [embeds the Weave manifest as a constant](https://github.com/kubernetes-sigs/kind/blob/4119e0bb893fd32b85883c1b7cecaab73cb71a54/pkg/build/node/const.go).

Reply:
The weave constant is not per-kubernetes version. This one is a bit more problematic.

Reply:
Downloading it separately is also not super ideal, will have to poke around the tarballs a bit. There should be a way to consistently obtain this 🤔

Reply:
The TODO about our own config is stale and wrong but the overall situation for these has not been good :(

Reply:
Hi @BenTheElder,

> The weave constant is not per-kubernetes version. This one is a bit more problematic.

And yet I also cannot find the affected files in any of the k/k repos except for release. So while they may be specific to the k/k version, they're not seemingly available per build. I initially wanted to use the version grokked from `kubernetes.tar.gz` and get the service and kubadm conf file specific for that version. Then I realized those were not available.

Reply:
> And yet I also cannot find the affected files in any of the k/k repos except for release.

there are systemd unit files in kubernetes/kubernetes in the build directory, the non-versioned k/release files are considered a bug and have caused many problems (and they may change inbetween releases happening, so checking in a static version here may not be right)